modules/features2d/src/brisk.cpp

```diff
@@ -353,13 +353,19 @@ BRISK_Impl::generateKernel(const std::vector<float> &radiusList,
   const int rings = (int)radiusList.size();
   CV_Assert(radiusList.size() != 0 && radiusList.size() == numberList.size());
   points_ = 0; // remember the total number of points
+  double sineThetaLookupTable[n_rot_];
+  double cosThetaLookupTable[n_rot_];
   for (int ring = 0; ring < rings; ring++)
   {
     points_ += numberList[ring];
   }
+  for (size_t rot = 0; rot < n_rot_; ++rot)
+  {
+      sineThetaLookupTable[rot] = sin(double(rot) * 2 * CV_PI / double(n_rot_));
```

Comment:
please, use the following implementation instead of repetitive sin/cos invocation (formulas need to be double-checked):

```
double cosval = 1., sinval = 0.;
double dcos = cos(2*CV_PI/double(n_rot_)), dsin = sin(2*CV_PI/double(n_rot_));
for( size_t rot = 0; rot < n_rot_; ++rot)
{
      sineThetaLookupTable[rot] = sinval;
      cosThetaLookupTable[rot] = cosval;
      double t = sinval*dcos + cosval*dsin;
      cosval = cosval*dcos - sinval*dsin;
      sinval = t;
}
```

the error will be less than `1e-13`, i.e. very low, but this initialization is much faster

Reply:
Done - this improves the runtime case from ~0.047s -> 0.036s.

Reply:
adding an update to description to document this.